### PR TITLE
Upgrade wagtail to 3.x

### LINKED ISSUE #2874: Roles checkboxes aren't aligned with labels
**Describe the bug**

The "Roles" 

**To Reproduce**

Go to edit a user's "Roles" -- the checkbox labels are not aligned with the checkboxes.

**Expected behavior**

The labels should appear next to checkboxes.

**Priority**

- Low priority (annoying, would be nice to not see)

**Affected roles**

- Staff

**Screenshots**

image

**Desktop (please complete the following information):**
 - OS: MacOS 12.4
 - Browser: Firefox

**Additional context**

In order to fix this in Hypha we need to add a hook to wagtail admin to load custom CSS on admin pages.

From there I want to note that simply adding float left to the checkbox doesn't do the full trick because of the way Wagtail checkboxes are rendered in Fun and Exciting ways.  In particular they used a `-5px` top which messes with vertical alignment.  That has [since been changed](https://github.com/wagtail/wagtail/commit/5996c5dd29200090948edefafa825c4b7ac43f5a#diff-e1bd478fcf7bc69d113aabd3a6a1cb0e3294cb6bd28aba916f8c21a9c7f9be47L199) though there is still a 2px `top` which might still cause a pixel of net awkwardness.

I'll have a PR for this soon.

### LINKED ISSUE #2880: Update to Wagtail 3.x

### PR DESCRIPTION
**Pull request by theskumar**
Fixes #2880
Fixes #2874
Closes #2881 

- Update to wagtail 3.
- Rename BASE_URL to WAGTAILADMIN_BASE_URL
- use_json_field argument added to StreamField
- Change edit handlers to panels and also path to templates
- Make the test_activation_email_includes_link pass
- Use wagtail 3.0.1 and use robust testing for the activation email
- [Add migrations for StreamField, use_json=True](https://github.com/HyphaApp/hypha/pull/2934/commits/273c47eacf44be1c519705bf6cd644a12b0c1208)]
- [Update to wagtail-purge==0.2.0](https://github.com/HyphaApp/hypha/pull/2934/commits/3e8e66cafd4e127299ba49711346e55a321dfdd9)
- [Fix “FilteredFieldPanel” for wagtail 3.x](https://github.com/HyphaApp/hypha/pull/2934/commits/2b76b1b9041de38b4d827ba36b8321d6a9b5e563)
- [Fix “ReadOnlyPanel” and “ReadOnlyInlinePlanel” for wagtail 3.x](https://github.com/HyphaApp/hypha/pull/2934/commits/a9a497f98a70177f20b6bd9c852f313d3ef6b865)

TODOs:
- [x] Fix the test randomly failing
- [x] Review and fix wagtail admin HTML overrides

### CODE REVIEW

File: hypha/apply/api/v1/screening/tests/test_views.py [hunk 1]
```diff
@@ -129,13 +129,15 @@ def test_add_screening_status(self):
             data={'id': self.yes_screening_status.id}
         )
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
-        self.assertEqual(len(response.json()), self.submission.screening_statuses.count())
-        first_submission_screening_status = self.submission.screening_statuses.order_by('-id').last()
+        self.assertEqual(len(response.json()), 2)
+
+        first_submission_screening_status = self.yes_default_screening_status
```
Comment (sandeepsajan0) [inline hypha/apply/api/v1/screening/tests/test_views.py]: I think we need to revert this line. I understand what you are thinking here but we have a unique logic in its [views here](https://github.com/HyphaApp/hypha/blob/main/hypha/apply/api/v1/screening/views.py#L76) that is responsible to change the screening statuses ids of a submission.

File: hypha/apply/api/v1/screening/tests/test_views.py [hunk 2]
```diff
@@ -129,13 +129,15 @@ def test_add_screening_status(self):
...
         self.assertEqual(response.json()[0]['id'], first_submission_screening_status.id)
         self.assertEqual(response.json()[0]['title'], first_submission_screening_status.title)
         self.assertEqual(response.json()[0]['yes'], first_submission_screening_status.yes)
         self.assertEqual(response.json()[0]['default'], first_submission_screening_status.default)
-        self.assertEqual(response.json()[1]['id'], self.submission.screening_statuses.order_by('id').last().id)
+
+        self.assertEqual(response.json()[1]['id'], self.yes_screening_status.id)
```
Comment (sandeepsajan0) [inline hypha/apply/api/v1/screening/tests/test_views.py]: We need to change this as well, as per the above comment.

**Comment by theskumar** [comment]
The test_add_screening_status is failing for me locally. The reason seems to be there is no pre-defined ordering on the tables, but you are expecting an `ordering` in the returned response objects.

image

I’m not sure how it’s passing in the circle CI, but it’s failing on my machine. @frjo @sandeepsajan0

**Review by frjo — APPROVED** [review]
Excellent work! Deploying to test now.

**Pull request merged**